### ISSUE TITLE
Lexicographic ordering can seem a bit off

### ISSUE THREAD

**Issue by Chesire** [Chesire/LintRules#18]
When using the Lexicographic ordering lint rule, since it is completely based on lexicographic ordering it takes into account `:` and `.` characters, causing an odd order when having multiple libraries from the same person.

ex:
```gradle
 implementation 'com.chesire.lintrules:lint-gradle:1.1.0'
 implementation 'com.chesire.lintrules:lint-xml:1.1.0'
 implementation 'com.chesire:lifecyklelog:2.1.0'
```
is valid, whereas
```gradle
 implementation 'com.chesire:lifecyklelog:2.1.0'
 implementation 'com.chesire.lintrules:lint-gradle:1.1.0'
 implementation 'com.chesire.lintrules:lint-xml:1.1.0'
```
should be valid instead.

**Comment by Chesire** [comment]
```kotlin
        dependencyItems
            .lastOrNull()
            ?.onlyIf({
                first == property && second.compareTo(dependency, false) > 0
            }, {
                context.report(
                    LexicographicDependencies.issue,
                    context.getLocation(valueCookie),
                    LexicographicDependencies.message
                )
            })
```

Needs to be a little smarter when doing the `compareTo` section, to stop once it encounters a `:`

**Comment by Chesire** [comment]
This has been fixed by #21.
